.github/workflows/interface-unit-tests.yml

```
@@ -106,6 +128,22 @@ jobs:
             "default": ["3.10"]
           }
           EOF
+          elif [ "${{ inputs.python_warning_level }}" == "error" ];
+          then
+            cat >python_versions.json <<-EOF
+          {
+            "default": ["3.10"],
+            "torch-tests": ["3.10"],
+            "tf-tests": ["3.10"],
+            "jax-tests": ["3.10"],
+            "all-interfaces-tests": ["3.10"],
+            "external-libraries-tests": ["3.10"],
+            "qcut-tests": ["3.10"],
+            "qchem-tests": ["3.10"],
+            "gradients-tests": ["3.10"],
+            "data-tests": ["3.10"],
+            "device-tests": ["3.10"]          }
```

```suggestion
          {
            "default": ["3.10"]       
          }
```

Agree with @mudit2812 here, all jobs use the default version unless we want a specific jobs to use a different version. So once we want to test against 3.12 we can just update it here once

---

Sounds good.